src/Symfony/Component/Workflow/Transition.php

```diff
@@ -39,18 +46,56 @@ public function getName(): string
     }
 
     /**
+     * @deprecated since Symfony 7.3, use getFromArcs() instead
+     *
      * @return string[]
      */
     public function getFroms(): array
     {
-        return $this->froms;
+        trigger_deprecation('symfony/workflow', '7.3', 'Method "%s()" is deprecated, use "getFromArcs()" instead.', __METHOD__);
+
+        return array_map(static fn (Arc $arc): string => $arc->place, $this->fromArcs);
+    }
+
+    /**
+     * @return Arc[]
+     */
+    public function getFromArcs(): array
+    {
+        return $this->fromArcs;
     }
 
     /**
+     * @deprecated since Symfony 7.3, use getFromArcs() instead
+     *
      * @return string[]
      */
     public function getTos(): array
     {
-        return $this->tos;
+        trigger_deprecation('symfony/workflow', '7.3', 'Method "%s()" is deprecated, use "getToArcs()" instead.', __METHOD__);
+
+        return array_map(static fn (Arc $arc): string => $arc->place, $this->toArcs);
+    }
+
+    /**
+     * @return Arc[]
+     */
+    public function getToArcs(): array
+    {
+        return $this->toArcs;
+    }
+
+    // No type hint for $arc to avoid implicit cast
+    private function normalize(mixed $arc): Arc
+    {
+        if ($arc instanceof Arc) {
+            return $arc;
+        }
+
+        if (is_string($arc)) {
+            return new Arc($arc, 1);
+        }
+
+        throw new \InvalidArgumentException(sprintf('The type of arc is invalid. Expected string or Arc, got "%s".', get_debug_type($arc)));
```

Comment:
I suggest using `TypeError` so that we get the same error type than if we use a parameter type `Arc|string` in the future